include/boost/pool/pool.hpp

```diff
@@ -295,6 +295,18 @@ class pool: protected simple_segregated_storage < typename UserAllocator::size_t
     //! Called if malloc/ordered_malloc needs to resize the free list.
     void * malloc_need_resize(); //! Called if malloc needs to resize the free list.
     void * ordered_malloc_need_resize();  //! Called if ordered_malloc needs to resize the free list.
+    void advance_next(size_type partition_size)
+    {
+      BOOST_USING_STD_MIN();
+      size_type nnext_size;
+      if(!max_size)
+        nnext_size = next_size << 1;
+      else if(next_size*partition_size/requested_size < max_size)
+        nnext_size = min BOOST_PREVENT_MACRO_SUBSTITUTION(next_size << 1, max_size * requested_size / partition_size);
+      else
+        return;
+      next_size = min BOOST_PREVENT_MACRO_SUBSTITUTION(nnext_size, max_chunks());
```

Comment:
`set_next_size` sets more than next_size, should we be calling `set_next_size(nnext_size)` here?

Reply:
But that's exactly what the bug is about. I added the bug description to the commit message and the PR now.